src/console/Microsoft.Health.Fhir.Ingest.Console.Common/StartupBase.cs

```diff
@@ -0,0 +1,159 @@
+﻿// --------------------------------------------------------------------------
+// <copyright file="StartupBase.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+// --------------------------------------------------------------------------
+
+using Azure.Messaging.EventHubs;
+using Azure.Storage.Blobs;
+using DevLab.JmesPath;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Health.Common.Auth;
+using Microsoft.Health.Common.Storage;
+using Microsoft.Health.Events.Common;
+using Microsoft.Health.Events.EventCheckpointing;
+using Microsoft.Health.Events.EventConsumers;
+using Microsoft.Health.Events.EventConsumers.Service;
+using Microsoft.Health.Events.EventHubProcessor;
+using Microsoft.Health.Events.EventProducers;
+using Microsoft.Health.Expressions;
+using Microsoft.Health.Fhir.Ingest.Template;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Fhir.Ingest.Console.Common
+{
+    public abstract class StartupBase
+    {
+        protected StartupBase()
+        {
+            Configuration = GetEnvironmentConfig();
+            ServiceCollection = new ServiceCollection();
+        }
+
+        public ServiceCollection ServiceCollection { get; set; }
+
+        public IConfiguration Configuration { get; }
+
+        public abstract string ApplicationType { get; }
+
+        public abstract string OperationType { get; }
+
+        public static IConfiguration GetEnvironmentConfig()
+        {
+            IConfiguration config = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.json", true, true)
+                .AddJsonFile("local.appsettings.json", true, true)
+                .AddEnvironmentVariables()
+                .Build();
+
+            return config;
+        }
+
+        public virtual void ConfigureServices(IServiceCollection services)
+        {
+            AddContentTemplateFactories(services);
+
+            services.AddSingleton<IEventProcessorClientFactory, EventProcessorClientFactory>();
```

Comment:
I think it would be good to refactor these into extension methods that can be grouped into functional categories.  It would make it easier to reuse specific pieces if needed.

Reply:
updated to use extension methods